examples/deck/example.js

```diff
@@ -73,59 +120,88 @@ export default class Example extends Component<
+      // a polygon was clicked
+      this.setState({ editingFeatureIndex: info.index });
+    } else {
+      console.log('deselect editing feature'); // eslint-disable-line
+      // open space was clicked, so stop editing
+      this.setState({ editingFeatureIndex: null });
+    }
+  };
+
   _resize = () => {
     this.forceUpdate();
   };
 
-  _toNebulaFeatureJunc(junc: Object) {
-    const geojson = {
-      type: 'Feature',
-      geometry: {
-        type: 'Point',
-        coordinates: junc.position
-      },
-      properties: null
-    };
-    const style = {
-      pointRadiusMeters: 20,
-      outlineRadiusMeters: 20,
-      fillColor: [1, 0, 0, 1],
-      outlineColor: [0, 0, 1, 1]
+  render() {
+    const { testFeatures } = this.state;
+
+    const viewport = {
+      ...this.state.viewport,
+      height: window.innerHeight,
+      width: window.innerWidth
     };
-    return new Feature(geojson, style);
-  }
 
-  render() {
-    const { editableJunctionsLayer, state } = this;
-    let { viewport } = state;
+    const editablePolygonsLayer = new EditablePolygonsLayer({
+      data: testFeatures,
+      editingFeatureIndex: this.state.editingFeatureIndex,
+      pickable: true,
+
+      onStartDraggingPoint: ({ coordinateIndexes }) => {
+        // eslint-disable-next-line no-console, no-undef
+        console.log(`Start dragging point`, coordinateIndexes);
+      },
+      onDraggingPoint: ({ feature, featureIndex, coordinateIndexes }) => {
+        // Immutably replace the feature being edited in the featureCollection
+        this.setState({
+          testFeatures: {
+            ...this.state.testFeatures,
+            features: [
+              ...this.state.testFeatures.features.slice(0, featureIndex),
+              feature,
+              ...this.state.testFeatures.features.slice(featureIndex + 1)
+            ]
+          }
+        });
+      },
+      onStopDraggingPoint: ({ coordinateIndexes }) => {
+        // eslint-disable-next-line no-console, no-undef
+        console.log(`Stop dragging point`, coordinateIndexes);
+      },
 
-    const { innerHeight: height, innerWidth: width } = window;
-    viewport = Object.assign(viewport, { height, width });
+      // Can specify GeoJsonLayer props
+      getFillColor: () => [0x00, 0x20, 0x70, 0x30],
+      getLineColor: () => [0x00, 0x20, 0x70, 0xc0],
+      getLineWidth: () => 30,
+      lineWidthMinPixels: 2,
+      lineWidthMaxPixels: 10,
 
-    const wmViewport = new WebMercatorViewport(viewport);
+      // As well as point layer props
+      getPointColor: () => [0x00, 0x20, 0x70, 0xff],
+      pointHighlightColor: [0xff, 0xff, 0xff, 0xff]
+    });
 
     return (
       <div style={styles.mapContainer}>
         <link href="https://api.mapbox.com/mapbox-gl-js/v0.44.0/mapbox-gl.css" rel="stylesheet" />
-        <MapGL {...viewport} onChangeViewport={this._onChangeViewport}>
+        <StaticMap {...viewport}>
```

Comment:
Why static map? Can you pan/zoom?

Reply:
If you let `Deck` manage the event handling, you can have static map as child of deck.

Reply:
I get this error if I place `StaticMap` inside `DeckGL` (using 5.2.1):
```
assert.js:6 Uncaught Error: MapController is no longer a React component
    at assert (assert.js:6)
    at new MapController (map-controller.js:73)
    at Deck._createController (deck.js:258)
    at new Deck (deck.js:111)
```

Reply:
5.3 is your friend. also you may want to add `controller=MapController`. This can totally be done in a second PR. See [deck.gl examples](https://github.com/uber/deck.gl/blob/master/examples/website/3d-heatmap/app.js#L164)